src/common/server.ts

```diff
@@ -142,6 +147,26 @@ async function createServer(
   return new LanguageClient(serverId, serverName, serverOptions, clientOptions);
 }
 
+async function useNative(settings: ISettings): Promise<boolean> {
+  switch (settings.languageServer) {
+    case "automatic":
+      let version = await getRuffVersion(settings);
+      for (let i = 0; i < 3; i++) {
+        if (version[i] > MIN_NATIVE_VERSION[i]) {
+          return true;
+        }
+        if (version[i] < MIN_NATIVE_VERSION[i]) {
+          return false;
+        }
```

Comment:
I think we should also consider what settings they are using and **not** enable the new server when they use any incompatible settings.

Reply:
I could see that work too. But I think we at least need warnings in that case that inform users that some of their settings were ignored.

Reply:
Considering how we deprecate other features, I think we should actually consider settings here as well, at least for now. Our normal deprecation procedure is:

* Deprecate the feature, warn about usage, but it remains functioning
* Remove the feature, but make it a hard error
* Remove the feature entirely

In that sense, I would expect that the extension keeps functioning without any changes if I use "old" settings but that it would show me a deprecation warning.

Reply:
Ok, I agree that we should look at the settings as well. Now, as the user explicitly opted into using the "auto" behavior, we should be providing warning.